test/unit/test.utils.js

```diff
@@ -27,20 +28,63 @@ describe("lib/utils", function () {
     process.chdir(prevDir);
   });
 
-  it("getManifest() returns manifest in cwd()", function (done) {
-    process.chdir(simpleAddonPath);
-    utils.getManifest().then(function(manifest) {
-      expect(manifest.name).to.be.equal("simple-addon");
-      expect(manifest.title).to.be.equal("My Simple Addon");
-      done();
+  describe("getManifest", function() {
+    it("returns manifest in cwd()", function (done) {
+      process.chdir(simpleAddonPath);
+      utils.getManifest().then(function(manifest) {
+        expect(manifest.name).to.be.equal("simple-addon");
+        expect(manifest.title).to.be.equal("My Simple Addon");
+        done();
+      }).catch(done);
+    });
+
+    it("returns manifest from custom XPI file", function (done) {
+      // Change into a non-package path.
+      process.chdir(path.join(__dirname, "..", "addons"));
+
+      utils.getManifest(simpleAddonXPI).then(function(manifest) {
+        expect(manifest.name).to.be.equal("simple-addon");
+        expect(manifest.title).to.be.equal("My Simple Addon");
+        done();
+      }).catch(done);
+    });
+
+    it("throws error for non-existant XPI file", function (done) {
```

Comment:
These tests look good but I think you can return a promise in chai if you want instead of using `done`. Works for me either way.

```js
it("throws error for non-existant XPI file", function () {
  var badXPIFile = path.join(__dirname, "..", "not-a-real-file.xpi");
  return utils.getManifest(badXPIFile).then(function() {
    chai.fail('Unexpected success');
  }).catch(function(err) {
    expect(err.message).to.include('ENOENT');
    expect(err.message).to.include(badXPIFile);
 });
});